src/precompiles/sgxattest.rs

```diff
@@ -151,6 +158,32 @@ fn sgxattest_random(_input: &[u8], gas_limit: u64) -> PrecompileResult {
     }
 }
 
+fn sgxattest_httpscall(_input: &[u8], gas_limit: u64) -> PrecompileResult {
+    let gas_used = 10000 as u64;
+    if gas_used > gas_limit {
+        return Err(PrecompileError::OutOfGas);
+    } else {
+
+        let url = String::from_utf8_lossy(_input).into_owned();
+        // let url = "https://dkg.alliance.dev";
+
+        // Decode the ABI input
+        // let decoded = decode(&[ParamType::String], _input).expect("failed to decode");
+        // let url = decoded[0].clone().into_string().expect("failed to decode");
+
+        // Perform the HTTP call
+        let client = Client::new();
+        let response = client.get(url).send().expect("failed to decode");
+        let response_body = response.text().expect("failed to decode");
+
```

Comment:
Do the errors that happen there due to decoding or because .send() failed for a different reason?
Same for the .text(), although in this case, it might be failed to parse/decode

Reply:
probably the expect string got copy/pasted, would have to come up with an appropriate string there, if an exception is even appropriate at all here